pkg/util/chunk/chunk.go

```diff
@@ -187,12 +187,24 @@ func (c *Chunk) MemoryUsage() (sum int64) {
 		return 0
 	}
 	for _, col := range c.columns {
-		curColMemUsage := int64(unsafe.Sizeof(*col)) + int64(cap(col.nullBitmap)) + int64(cap(col.offsets)*8) + int64(cap(col.data)) + int64(cap(col.elemBuf))
-		sum += curColMemUsage
+		sum += int64(unsafe.Sizeof(*col)) + int64(cap(col.nullBitmap)) + int64(cap(col.offsets)*8) + int64(cap(col.data)) + int64(cap(col.elemBuf))
 	}
 	return
 }
 
+// GetMemoryUsageCap returns the total memory usage that may be used in the future.
+// We ignore the size of Column.length and Column.nullCount
+// since they have little effect of the total memory usage.
+func (c *Chunk) GetMemoryUsageCap(capacity int) (sum int64) {
```

Comment:
Why not use chunk.MemoryUsage, what's the difference?

Reply:
> Why not use chunk.MemoryUsage, what's the difference?

`chunk.MemoryUsage` returns the memory that has been allocated. However, we don't allocate any memory when the chunk is initialized. So, we need a new function to return the memory that the chunk may be allocated in the future.